src/Scripting/Core/ScriptOptions.cs

```diff
@@ -282,5 +301,17 @@ private static MetadataReference CreateReferenceFromAssembly(Assembly assembly)
         /// <exception cref="ArgumentNullException"><paramref name="imports"/> is null or contains a null reference.</exception>
         public ScriptOptions AddImports(params string[] imports) =>
             AddImports((IEnumerable<string>)imports);
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with debugging information enabled.
+        /// </summary>
+        public ScriptOptions WithEmitDebugInformation(bool emitDebugInformation) =>
+            emitDebugInformation == EmitDebugInformation ? this : new ScriptOptions(this) { EmitDebugInformation = emitDebugInformation };
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="FileEncoding"/>.
+        /// </summary>
+        public ScriptOptions WithFileEncoding(Encoding encoding) => 
```

Comment:
sure, so the suggestion - if I understand correctly - is to rely on letting encoding be detected by `SourceText.From(stream)` because internally it would try to do it by looking at BOM etc? 

When we add the stream overloads, do you also remove manual encoding setting from options and specify that using the non-stream scripting methods will not allow debugging (since under the hood it will use non-debuggable `SourceText`)?